src/main/java/services/impl/FruitDaoServiceImp.java

```diff
@@ -0,0 +1,46 @@
+package services.impl;
+
+import db.Storage;
+import java.util.HashSet;
+import java.util.Set;
+import model.Fruit;
+import model.FruitRecord;
+import services.FruitDaoService;
+
+public class FruitDaoServiceImp implements FruitDaoService {
+    private Storage storage = new Storage();
+
+    public FruitDaoServiceImp(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void put(FruitRecord fruitRecord) {
+        Set<Fruit> fruitsInStorage = storage.getFruitsInStorage();
```

Comment:
Also you may not understand that `getFruitsInStorage()` returns a link of the object and you can modify that object without creating a local variable of it, that is an advise for the future :)